addons/website/static/src/snippets/s_map/000.js

```diff
@@ -0,0 +1,41 @@
+/** @odoo-module **/
+
+var publicWidget = require('web.public.widget');
+
+publicWidget.registry.Map = publicWidget.Widget.extend({
+    selector: '.s_map',
+
+    /**
+     * @override
+     */
+    start() {
+        if (!this.el.querySelector('.s_map_embedded')) {
+            // The iframe is not found inside the snippet. This is probably due
+            // the sanitization of a field during the save, like in a product
+            // description field.
+            // In such cases, reconstruct the iframe.
+            const dataset = this.$target[0].dataset;
+            if (dataset.mapAddress) {
+                const url = 'https://maps.google.com/maps?q=' + encodeURIComponent(dataset.mapAddress)
+                    + '&t=' + encodeURIComponent(dataset.mapType)
+                    + '&z=' + encodeURIComponent(dataset.mapZoom)
+                    + '&ie=UTF8&iwloc=&output=embed';
```

Comment:
Either this or we just save that URL as data-attribute as you suggested and just use it here.

The current way has 2 advantages:
- It doesn't need a change in the options.js (we don't really care tho)
- It will work for already drag & dropped not working s_map (shouldn't be much tho)

Reply:
> Either this or we just save that URL as data-attribute as you suggested

Nah, that was to better illustrate the suggestion, I would never prefer that, duplicated data :wink: 

> It doesn't need a change in the options.js

Actually, I would prefer options.js to be changed to not have duplicated code for this URL generation. It's kinda a strange case. Normally what is created in the `start` of a public widget should be removed in its `destroy`... but in this case I guess it is not what we really want. What could be done however is letting the public widget always be the one in charge of setting the right src (so the options.js would not set it, just ask for a refresh of the widget (automatically done anyway if `disabledInEditableMode` is set to `false` on the widget). If not that, then I guess some kind of utils method in website/utils.js called by both the option and the widget (I guess that would be the most stable way so I would do that) -> Whichever is fine by me but I hate duplicated code :)

> It will work for already drag & dropped not working s_map

Actually, that would be a reason not to do this as you might "wake up" map snippets that user though were removed but we should not consider that side-effect IMO.

Reply:
> I will go for a shared utility method then, I want the 000.js to be used only when the map is inside a sanitized field which removed the iframe element (see the first condition and comment if (!this.el.querySelector('.s_map_embedded')) {).

My other suggestion would not prevent to do that as well, but the utils method is probably the right stable approach indeed :+1: 

> The less js to execute we need the better.

If code clarity stays the same, sure :+1: 

--

Note: in master, we should remove the iframe from the snippet definition and let your utility method create it then (and thus both called by the widget and the option). That would prevent duplicated code (if you want to add an attribute on the iframe, you don't have to think about adding it on the snippet xml and the 000.js)